system/Encryption/Encryption.php

```diff
@@ -99,12 +99,11 @@ class Encryption
 	 */
 	public function __construct(BaseConfig $config = null)
 	{
-		if (empty($config))
-		{
-			$config = new \Config\Encryption();
-		}
-		$this->driver = $config->driver;
+		$config = $config ?? new \Config\Encryption();
+
 		$this->key    = $config->key;
+		$this->driver = $config->driver;
+		$this->digest = $config->digest;
```

Comment:
probably add default value:

```suggestion
              $this->digest = $config->digest ?? 'SHA512';
```